src/lib.rs

```diff
@@ -126,23 +127,69 @@ impl Debug for GetResult {
 impl GetResult {
     /// Collects the data into a [`Bytes`]
     pub async fn bytes(self) -> Result<Bytes> {
-        collect_bytes(self.into_stream(), None).await
+        match self {
+            Self::File(mut file, path) => {
+                maybe_spawn_blocking(move || {
+                    let len = file
+                        .seek(SeekFrom::End(0))
```

Comment:
will seeking the same underlying file descriptor cause issues if this code is used concurrently?

Reply:
Yes, although you need a mutable reference to File to seek so you would have to be using something like TryClone, which I'd argue is not something we need to protect against.